test/framework/clusterctl/repository.go

```diff
@@ -134,9 +143,44 @@ func CreateRepository(ctx context.Context, input CreateRepositoryInput) string {
 	}
 	clusterctlConfigFile.write()
 
+	// creates a clusterctl config file to be used for working with such repository with only the providers supported in clusterctl < v1.3
+	clusterctlConfigFileV1_2 := &clusterctlConfig{
+		Path: filepath.Join(input.RepositoryFolder, clusterctlConfigV1_2FileName),
+		Values: map[string]interface{}{
+			"providers":       providersV1_2,
+			"overridesFolder": overridePath,
+		},
+	}
+	for key := range input.E2EConfig.Variables {
+		clusterctlConfigFileV1_2.Values[key] = input.E2EConfig.GetVariable(key)
+	}
+	clusterctlConfigFileV1_2.write()
+
 	return clusterctlConfigFile.Path
 }
 
+// AdjustConfigPathForBinary adjusts the clusterctlConfigPath in case the clusterctl version v1.3.
+func AdjustConfigPathForBinary(clusterctPath, clusterctlConfigPath string) string {
+	clusterctl := exec.NewCommand(
+		exec.WithCommand(clusterctPath),
+		exec.WithArgs("version", "--output", "short"),
+	)
+	stdout, stderr, err := clusterctl.Run(context.Background())
+	if err != nil {
+		Expect(err).ToNot(HaveOccurred(), "failed to run clusterctl version:\nstdout:\n%s\nstderr:\n%s", string(stdout), stderr)
+	}
+	data := stdout
+	version, err := semver.ParseTolerant(string(data))
+	if err != nil {
+		Expect(err).ToNot(HaveOccurred(), "clusterctl version returned an invalid version: %s (%v)", string(data), err)
```

Comment:
```suggestion
		Expect(err).ToNot(HaveOccurred(), "clusterctl version returned an invalid version: %s", string(data))
```
Just to avoid printing the error twice (gomega will already do that)